src/mapnik_image.cpp

```diff
@@ -1620,6 +1644,131 @@ void Image::EIO_AfterOpen(uv_work_t* req)
+    NanReturnValue(_fromSVGSync(false, args));
+}
+
+// Read from a file
+NAN_METHOD(Image::fromSVGSync)
+{
+    NanScope();
+    NanReturnValue(_fromSVGSync(true, args));
+}
+
+Local<Value> Image::_fromSVGSync(bool fromFile, _NAN_METHOD_ARGS)
+{
+    NanEscapableScope();
+
+    if (!fromFile && (args.Length() < 1 || !args[0]->IsObject())) {
+        NanThrowTypeError("must provide a buffer argument");
+        return NanEscapeScope(NanUndefined());
+    }
+
+    if (fromFile && (args.Length() < 1 || !args[0]->IsString())) {
+        NanThrowTypeError("must provide a filename argument");
+        return NanEscapeScope(NanUndefined());
+    }
+
+
+    double scale = 1.0;
+    if (args.Length() >= 2) {
+        if (!args[1]->IsObject()) {
+            NanThrowTypeError("optional second arg must be an options object");
+            return NanEscapeScope(NanUndefined());
+        }
+        Local<Object> options = args[1]->ToObject();
+        if (options->Has(NanNew("scale")))
+        {
+            Local<Value> scale_opt = options->Get(NanNew("scale"));
+            if (!scale_opt->IsNumber()) {
+                NanThrowTypeError("'scale' must be an number");
+                return NanEscapeScope(NanUndefined());
+            }
+            scale = scale_opt->NumberValue();
+        }
+    }
+
+    try
+    {
+
+        //std::shared_ptr<mapnik::marker const> marker = mapnik::marker_cache::instance().find(svg_filename, false);
+
+        using namespace mapnik::svg;
+        mapnik::svg_path_ptr marker_path(std::make_shared<mapnik::svg_storage_type>());
+        vertex_stl_adapter<svg_path_storage> stl_storage(marker_path->source());
+        svg_path_adapter svg_path(stl_storage);
+        svg_converter_type svg(svg_path, marker_path->attributes());
+        svg_parser p(svg);
+        if (fromFile)
+        {
+            p.parse(TOSTR(args[0]));
+        }
+        else
+        {
+            Local<Object> obj = args[0]->ToObject();
+            if (obj->IsNull() || obj->IsUndefined() || !node::Buffer::HasInstance(obj)) {
+                NanThrowTypeError("first argument is invalid, must be a Buffer");
+                return NanEscapeScope(NanUndefined());
+            }
+            std::string svg_buffer(node::Buffer::Data(obj),node::Buffer::Length(obj));
+            p.parse_from_string(svg_buffer);
+        }
+
+        double lox,loy,hix,hiy;
+        svg.bounding_rect(&lox, &loy, &hix, &hiy);
+        marker_path->set_bounding_box(lox,loy,hix,hiy);
+        marker_path->set_dimensions(svg.width(),svg.height());
+
+        using pixfmt = agg::pixfmt_rgba32_pre;
+        using renderer_base = agg::renderer_base<pixfmt>;
+        using renderer_solid = agg::renderer_scanline_aa_solid<renderer_base>;
+        agg::rasterizer_scanline_aa<> ras_ptr;
+        agg::scanline_u8 sl;
+
+        double opacity = 1;
+        int w = svg.width() * scale;
+        int h = svg.height() * scale;
+
+        mapnik::image_rgba8 im(w, h);
```

Comment:
Should there be a check to see if this is zero and throw if this occurs, otherwise bad things could occur.